src/equations/compressible_euler_multicomponent_2d.jl

```diff
@@ -7,61 +7,58 @@
 
 Multicomponent version of the compressible Euler equations for an ideal gas in two space dimensions.
 """
-struct CompressibleEulerMulticomponentEquations2D{RealT<:Real} <: AbstractCompressibleEulerMulticomponentEquations{2, 5}
-  gamma1          ::RealT 
-  gamma2          ::RealT
-  gas_constant1   ::RealT
-  gas_constant2   ::RealT
-  cv1             ::RealT
-  cv2             ::RealT
-  cp1             ::RealT
-  cp2             ::RealT
-end
 
-function CompressibleEulerMulticomponentEquations2D()
+struct CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS} <: AbstractCompressibleEulerMulticomponentEquations{2, NVARS}
+  gamma             ::SVector{NCOMP, Real} 
+  gas_constant      ::SVector{NCOMP, Real}
+  cv                ::SVector{NCOMP, Real}
+  cp                ::SVector{NCOMP, Real}
+  CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS}(gamma, gas_constant) where {NCOMP, NVARS} = new(gamma, gas_constant, gas_constant./(gamma.-1), gas_constant + gas_constant./(gamma .-1))
+end
 
-  # Set ratio of specific heat of the gas per species 
-  # (It holds: gamma = cp/cv) (Can be used for consistency check!)
-  gamma1  = 1.4
-  gamma2  = 1.4
-  
-  # Set specific gas constant with respect to the molar mass per species 
-  # (It holds: gas_constant = R/M, with R = molar gas constant, M = molar mass) 
-  # (For a calorically and thermally perfect gas Mayer's relation holds: gas_constant = cp - cv) (Can be used for consistency check!)
-  gas_constant1     = 0.4
-  gas_constant2     = 0.4
 
+function CompressibleEulerMulticomponentEquations2D(; gamma::AbstractVector{<:Real}, gas_constant::AbstractVector{<:Real})
```

Comment:
Here, I would prefer to not restrict the types of `gamma` and `gas_constant` to `AbstractVector{<:Real}`, since this will not allow passing tuples. As far as I know, passing tuples would also be better for performance and type stability since the length of tuples (and hence `NCOMPS, NVARS`) can be determined at compile time. Thus, you could insist on getting tuples of the same length here for `gamma` and `gas_constant`. Then, you could also remove the additional check below.

Reply:
I removed the restrictions in c0314d0.